ui/src/app/archived-workflows/components/archived-workflow-list/archived-workflow-list.tsx

```diff
@@ -122,65 +106,61 @@ export class ArchivedWorkflowList extends BasePage<RouteComponentProps<any>, Sta
         }
-        const url = 'archived-workflows/' + namespace + '?' + params.toString();
-        history.pushState(null, '', uiUrl(url));
-        this.fetchArchivedWorkflows(namespace, selectedPhases, selectedLabels, minStartedAt, maxStartedAt, pagination);
+        return params;
+    }
+
+    private saveHistory() {
+        this.url = uiUrl('archived-workflows/' + this.state.namespace + '?' + this.filterParams.toString());
+        Utils.setCurrentNamespace(this.state.namespace);
     }
 
     private fetchArchivedWorkflows(namespace: string, selectedPhases: string[], selectedLabels: string[], minStartedAt: Date, maxStartedAt: Date, pagination: Pagination): void {
-        let archivedWorkflowList;
-        let newNamespace = namespace;
-        if (!this.state.initialized) {
-            archivedWorkflowList = services.info.getInfo().then(info => {
-                if (info.managedNamespace) {
-                    newNamespace = info.managedNamespace;
-                }
-                this.setState({initialized: true, managedNamespace: !!info.managedNamespace});
-                return services.archivedWorkflows.list(newNamespace, selectedPhases, selectedLabels, minStartedAt, maxStartedAt, pagination);
-            });
-        } else {
-            if (this.state.managedNamespace) {
-                newNamespace = this.state.namespace;
-            }
-            archivedWorkflowList = services.archivedWorkflows.list(newNamespace, selectedPhases, selectedLabels, minStartedAt, maxStartedAt, pagination);
-        }
-        archivedWorkflowList
+        services.archivedWorkflows
+            .list(namespace, selectedPhases, selectedLabels, minStartedAt, maxStartedAt, pagination)
             .then(list => {
-                this.setState({
-                    namespace: newNamespace,
-                    workflows: list.items || [],
-                    selectedPhases,
-                    selectedLabels,
-                    minStartedAt,
-                    maxStartedAt,
-                    pagination: {
-                        limit: pagination.limit,
-                        offset: pagination.offset,
-                        nextOffset: list.metadata.continue
+                this.setState(
+                    {
+                        namespace,
+                        workflows: list.items || [],
+                        selectedPhases,
+                        selectedLabels,
+                        minStartedAt,
+                        maxStartedAt,
+                        pagination: {
+                            limit: pagination.limit,
+                            offset: pagination.offset,
+                            nextOffset: list.metadata.continue
+                        }
                     },
-                    loading: false
-                });
-                Utils.setCurrentNamespace(newNamespace);
+                    this.saveHistory
```

Comment:
part of the `saveHistory` pattern is to invoke it after only on successful state update, again - you'll see this as a repeating pattern